src/config.ts

```diff
@@ -586,6 +586,42 @@ export const ARNS_NAMES_CACHE_TTL_SECONDS = +env.varOrDefault(
   `${60 * 60}`, // 1 hour
 );
 
+// Controls the maximum time allowed for requests to AO for ARIO process state. By default, requests
+// should resolve in less than 3 seconds, but we set to 10 seconds to account for the worst case scenario.
+// If requests exceed this timeout, they will be considered failed and may trigger the circuit breaker if the error threshold is reached.
+export const ARIO_PROCESS_DEFAULT_CIRCUIT_BREAKER_TIMEOUT_MS =
+  +env.varOrDefault(
+    'ARIO_PROCESS_DEFAULT_CIRCUIT_BREAKER_TIMEOUT_MS',
+    `${10 * 1000}`, // 10 seconds
+  );
+
+// Controls the percentage of failed requests to AO for ARIO process state that will trigger
+// the circuit breaker to open. This is set to a relatively low threshold (30%)
+// to compensate for the extended timeout (10 seconds) configured above.
+export const ARIO_PROCESS_DEFAULT_CIRCUIT_BREAKER_ERROR_THRESHOLD_PERCENTAGE =
+  +env.varOrDefault(
+    'ARIO_PROCESS_DEFAULT_CIRCUIT_BREAKER_ERROR_THRESHOLD_PERCENTAGE',
+    '30', // 30% failure limit before circuit breaker opens
+  );
+
+// Defines the time window for tracking errors when retrieving ARIO process state from AO
+// The circuit breaker counts failures within this rolling time window to determine
+// if the error threshold percentage has been exceeded
+export const ARIO_PROCESS_DEFAULT_CIRCUIT_BREAKER_ROLLING_COUNT_TIMEOUT_MS =
```

Comment:
Is this enough time if we're waiting 2 minutes between refreshes?

Reply:
good call out -

if increased (i.e., adjusted to something like 10 minutes), this circuit breaker would trigger if 2/5 (40%) refreshes fail. the circuit breaker would then prevent requests to the CU for 10 minutes. 

seems reasonable enough? we may want to consider a longer backoff period (~20 minutes) as well - given CU issues tend to linger.

Reply:
Yeah, 2/5 seems good.

Reply:
I'm fine with a 20 minute back-off as well. Once we implement the better trusted gateway fallback that should be quite safe.

Reply:
updated to 10 min window and 20 minute back-off in [fc0e72d](https://github.com/ar-io/ar-io-node/pull/359/commits/fc0e72d62f5845c001f7ab7f6141fa2ab0ed320e)